src/jdk.compiler/share/classes/com/sun/tools/javac/code/Source.java

```diff
@@ -263,6 +263,7 @@ public enum Feature {
         FLEXIBLE_CONSTRUCTORS(JDK22, Fragments.FeatureFlexibleConstructors, DiagKind.NORMAL),
         MODULE_IMPORTS(JDK23, Fragments.FeatureModuleImports, DiagKind.PLURAL),
         PRIVATE_MEMBERS_IN_PERMITS_CLAUSE(JDK19),
+        COMPILE_TIME_TYPE_SIG_POLYMORPHIC_MTH(JDK24),
```

Comment:
Nit: I'd pick a name that is more representative of what the behavior is. E.g. `erase_poly_sig_return_type`. Same for the field in `Infer` storing the boolean flag.